Add example code for QuicTransport

vasilvv opened the pull request:
This includes a flexible client and a Python-based server.  Note that
the client code does not currently use the sample template, as it has
too many features.  I'll write a shorter version that fits into the
template later, but the full thing is also useful to have around.

vasilvv committed:
Add example code for QuicTransport

This includes a flexible client and a Python-based server.  Note that
the client code does not currently use the sample template, as it has
too many features.  I'll write a shorter version that fits into the
template later, but the full thing is also useful to have around.

beaufortfrancois commented:
@jeffposnick Can you review?

jeffposnick reviewed (changes requested):
Thanks a ton for contributing this, @vasilvv!

Using your own JS/CSS is totally fine.

I do have some feedback about making the sample server easier to run. It would make sense to put this in either the doc comment at the top of `quic_transport_server.py,` or alternatively to add it to the placeholder `README.md` that's currently checked in.

- Not everyone is going to know how to set up `aioquic`. Linking to https://github.com/aiortc/aioquic (or a better setup guide, if you know of one) would help.

- Creating `certificate.pem` and `certificate.key` is going to be a challenge for folks. I believe that [this page](https://www.chromium.org/quic/playing-with-quic) is the best guide for creating them, but honestly, as someone who's never checked out and built Chromium before, it's a big task. At the least we should link to that page, but if it's "safe" to check in a pre-generated `.pem` and `.key` pair that could only be used on `localhost`, perhaps we could do that?

- I know that you have a [doc](https://docs.google.com/document/d/1QPPC27POGoVg8TaxB3yhuuLQXtlUwbCQumT6gJHmR0I/edit#) explaining the use of the `--origin-to-force-quic-on`, which will be needed when running the server locally. Could that info be moved to either the `README.md` or the Python comment?

- Finally, we should sign this sample up for the Origin Trial and include that in a `<meta>` tag, but I don't see the trail active yet at https://developers.chrome.com/origintrials/#/trials/active

Let me know if you'd like my help with any of those steps.

vasilvv committed:
Add details on aioquic and how to handle certificates

vasilvv commented:
I fixed added some comments to the Python server.  I think this addresses all of your comments except the last one (which I've asked my colleagues about, but can't do until the origin trial is added to the dashboard).

jeffposnick reviewed (approved):
Thanks! I think those changes should help developers get started.

I can sign up for the OT and add in the `<meta>` tag in a follow-up PR once the OT is active.
